expression/builtin_other.go

```diff
@@ -776,7 +776,6 @@ func (b *builtinSetStringVarSig) evalString(row chunk.Row) (res string, isNull b
 	if err != nil {
 		return "", isNull, err
 	}
-	varName = strings.ToLower(varName)
```

Comment:
Will this break any implicit dependencies and introduce regression?

Reply:
We assume that the lower case is used, that's what we have done when getting the variables.
So wrapping strings.ToLower(varName) into SetUserVar would avoiding the missing converstion.

Reply:
The reason of introducing this bug is that:
1. first, convert to the lower-case name, then call SetUserVar.
2. when supporting user variable for load statement, call SetUserVar directly.

So I move the `convert to the lower-case name` into SetUserVar.